blackjax/hmc.py

```diff
@@ -50,26 +64,61 @@ class HMCInfo(NamedTuple):
     num_integration_steps: int
 
 
-new_state = base.new_hmc_state
-
-
-def kernel(
+def hmc(
     logprob_fn: Callable,
     step_size: float,
     inverse_mass_matrix: Array,
     num_integration_steps: int,
     *,
     integrator: Callable = integrators.velocity_verlet,
     divergence_threshold: int = 1000,
+) -> SamplingAlgorithm:
+
+    # It has an euclidean metric by default
+    kernel = hmc_kernel(integrator, divergence_threshold)
+
+    def init_fn(position: PyTree):
+        return jax.jit(hmc_init, static_argnums=(1,))(position, logprob_fn)
+
+    def step_fn(rng_key: PRNGKey, state: HMCState) -> Tuple[HMCState, HMCInfo]:
+        # `np.ndarray` and `DeviceArray`s are not hashable and thus cannot be used as static arguments.`
+        # Workaround: https://github.com/google/jax/issues/4572#issuecomment-709809897
+        kernel_fn = jax.jit(
+            kernel,
+            static_argnums=(2, 3, 5),
+        )
+        return kernel_fn(
+            rng_key,
+            state,
+            logprob_fn,
+            step_size,
+            inverse_mass_matrix,
+            num_integration_steps,
+        )
+
+    return SamplingAlgorithm(init_fn, step_fn)
+
+
+def hmc_init(position: PyTree, logprob_fn: Callable):
+    def potential_fn(x):
+        return -logprob_fn(x)
+
+    potential_energy, potential_energy_grad = jax.value_and_grad(potential_fn)(position)
+    return HMCState(position, potential_energy, potential_energy_grad)
+
+
+def hmc_kernel(
```

Comment:
Could you add some test case of calling `hmc_kernel` directly? (Same for NUTS and RMH)